### PR TITLE
mon: Apply mon store settings more efficiently with the assimilate conf

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The monstore has a `SetAll()` method for applying multiple settings that was applying them individually. Now we use the assimilate config option to make this more efficient.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/config/monstore.go [hunk 1]
```diff
@@ -207,3 +191,64 @@ func (m *MonStore) SetKeyValue(key, value string) error {
...
+
+	err = ioutil.WriteFile(assimilateConfPath.Name(), []byte(""), 0600)
+	if err != nil {
+		rook.TerminateFatal(errors.Wrapf(err, "failed to write config file"))
+	}
+
+	defer func() {
+		err := os.Remove(assimilateConfPath.Name())
+		if err != nil {
+			logger.Errorf("failed to remove file %q. %v", assimilateConfPath.Name(), err)
```
Comment (sp98) [inline pkg/operator/ceph/config/monstore.go:209]: should this error be returned?

File: pkg/operator/ceph/config/monstore.go [hunk 2]
```diff
@@ -207,3 +191,64 @@ func (m *MonStore) SetKeyValue(key, value string) error {
 	}
 	return nil
 }
+
+func (m *MonStore) SetAll(clientName string, settings map[string]string) error {
+	assimilateConfPath, err := ioutil.TempFile("", "")
```
Comment (sp98) [inline pkg/operator/ceph/config/monstore.go:196]: ioutil functions are mostly deprecated started go 1.16. The [doc](https://pkg.go.dev/io/ioutil@go1.18) is suggesting to use the `os` or `io` package.

File: pkg/operator/ceph/config/monstore.go [hunk 3]
```diff
@@ -207,3 +191,64 @@ func (m *MonStore) SetKeyValue(key, value string) error {
...
+
+	out, err := cephCmd.RunWithTimeout(exec.CephCommandsTimeout)
+	if err != nil {
+		logger.Errorf("failed to run command ceph %s", args)
+
+		fileContent, err := ioutil.ReadFile(assimilateConfPath.Name() + ".out")
+		if err != nil {
+			logger.Errorf("failed to open assimilate output file %s.out. %v", assimilateConfPath.Name(), err)
+		}
+		logger.Infof("failed to apply ceph settings:\n%s", string(fileContent))
```
Comment (sp98) [inline pkg/operator/ceph/config/monstore.go:246]: log as error?

**Review by sp98 — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/config/monstore.go (hunk 2 above)
Comment (travisn) [reply pkg/operator/ceph/config/monstore.go:196]: This code just moved, but good point to clean it up at the same time, thanks

**Review by subhamkrai — APPROVED** [review]
LGTM, apart form Santosh Comments

File: pkg/operator/ceph/config/monstore.go (hunk 1 above)
Comment (travisn) [reply pkg/operator/ceph/config/monstore.go:209]: We can't return an err from the defer function, otherwise agreed

**Pull request merged**